subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c

```diff
@@ -197,15 +197,26 @@ static void sync_rest_client_data(struct nrf_cloud_rest_context *const rest_ctx,
 
 static int do_rest_client_request(struct nrf_cloud_rest_context *const rest_ctx,
 	struct rest_client_req_context *const req,
-	struct rest_client_resp_context *const resp)
+	struct rest_client_resp_context *const resp, bool check_status_good)
 {
 	int ret = rest_client_request(req, resp);
 
 	sync_rest_client_data(rest_ctx, req, resp);
 
-	return ret;
-}
+	if (ret) {
+		LOG_DBG("REST client request failed with error code %d", ret);
+		return ret;
+	} else if (rest_ctx->status == NRF_CLOUD_HTTP_STATUS_NONE) {
+		LOG_DBG("REST request endpoint closed connection without reply.");
+		return -ESHUTDOWN;
```

Comment:
perhaps add a `@note` to the `keep_alive` parameter in the `nrf_cloud_rest_context` struct that if a REST call 
 returns `-ESHUTDOWN`, it likely means the socket is closed and the user should handle it appropriately.
 
currently there is a `@note` indicating that the socket may be closed on an errant call.  i'm not sure if this is still the case, as the lib doesn't do any explicit disconnects.

Reply:
I agree

Although I don't think it should be specific to keep_alive, since we can get this error code even with keep_alive disabled.

This makes me wonder if it would be worthwhile to add an error codes section and then reference that section in all of the return value documentations for API call functions.

What do you think?

Reply:
ah right, yes then it does need to be mentioned in each function.
if there is a proper way to reference an error code section, that is fine.
otherwise it will have to be a `* @retval -ESHUTDOWN ...` for each function.

Reply:
Since we're updating documentation anyways, and since this is the last thing standing in the way of the documentation changes I've been wanting to do, I'm going to go ahead and integrate all of the doc changes simultaneously as part of this issue.